symforce/slam/imu_preintegration/imu_factor.h

```diff
@@ -0,0 +1,82 @@
+/* ----------------------------------------------------------------------------
+ * SymForce - Copyright 2022, Skydio, Inc.
+ * This source code is under the Apache 2.0 license found in the LICENSE file.
+ * ---------------------------------------------------------------------------- */
+
+#pragma once
+
+#include <Eigen/Dense>
+
+#include <sym/pose3.h>
+
+#include "./preintegrated_imu_measurements.h"
+
+namespace sym {
+
+/**
+ * A Functor storing on-manifold preintegrated IMU measurements between two times and
+ * which computes the local-coordinate difference between the final state (pose and
+ * velocity) and the state you would expect given the initial state and
+ * the IMU measurements.
+ *
+ * Based heavily on the on manifold ImuFactor found in GTSAM and on the paper:
+ *
+ * Christian Forster, Luca Carlone, Frank Dellaert, and Davide Scaramuzza,
+ * “IMU Preintegration on Manifold for Efficient Visual-Inertial Maximum-a-Posteriori Estimation”,
+ * Robotics: Science and Systems (RSS), 2015.
```

Comment:
Is this a sufficient citation? Should I put it in other places as well?

Reply:
You could also cite the original paper by Lupton.